pipeline_utils/lib/yaml_parser.py

```diff
@@ -0,0 +1,473 @@
+                                error.validator,
+                                error.message,
+                                error.relative_path,
+                                error.schema
+                                )
+                            )
+            raise SchemaError
+
+
+###############################################################
+#   YAMLWorkflow, YAML Workflow
+###############################################################
+class YAMLWorkflow(YAMLTemplate):
+
+    def __init__(self, data):
+        """
+        """
+        super().__init__(data, yaml_workflow_schema)
+        # Validate data with schema
+        self._validate()
+        # Load attributes
+        for key, val in data.items():
+            setattr(self, key, val)
+
+    def _arguments_input(self):
+        """
+        """
+        arguments = []
+        for name, values in self.input.items():
+            type, format = values[self.ARGUMENT_TYPE_SCHEMA].split('.')
+            if type == 'file':
+                argument_type = 'Input file'
+            elif type == 'parameter':
+                argument_type = 'parameter'
+            argument_ = {
+                self.ARGUMENT_TYPE_SCHEMA: argument_type,
+                self.WORKFLOW_ARGUMENT_NAME_SCHEMA: name
+                }
+            arguments.append(argument_)
+
+        return arguments
+
+    def _arguments_output(self):
+        """
+        """
+        arguments = []
+        for name, values in self.output.items():
+            type, format = values[self.ARGUMENT_TYPE_SCHEMA].split('.')
+            if type == 'file':
+                argument_type = 'Output processed file'
+                argument_ = {
+                    self.ARGUMENT_FORMAT_SCHEMA: format,
+                    self.ARGUMENT_TYPE_SCHEMA: argument_type,
+                    self.WORKFLOW_ARGUMENT_NAME_SCHEMA: name,
+                    self.SECONDARY_FILE_FORMATS_SCHEMA: values.get(self.SECONDARY_FILES_SCHEMA, [])
+                }
+            elif type == 'qc':
+                argument_type = 'Output QC file'
+                argument_ = {
+                    self.ARGUMENT_TYPE_SCHEMA: argument_type,
+                    self.WORKFLOW_ARGUMENT_NAME_SCHEMA: name,
+                    'argument_to_be_attached_to': values['argument_to_be_attached_to'],
+                    'qc_type': format,
+                    'qc_zipped': values.get('zipped', False),
+                    'qc_html': values.get('html', False),
+                    'qc_json': values.get('json', False),
+                    'qc_table': values.get('table', False)
+                }
+            arguments.append(argument_)
+
+        return arguments
+
+    def to_json(
+               self,
+               VERSION='VERSION',
+               INSTITUTION='INSTITUTION', # alias
+               PROJECT='PROJECT', # alias
+               WFLBUCKET_URL='s3://WFLBUCKET/PIPELINE/VERSION',
```

Comment:
These kwargs should probably be lower-case snakecase, right? They're linted as constants when all caps here. Similar elsewhere but didn't mark all instances.
```suggestion
               version='VERSION',
               institution='INSTITUTION', # alias
               project='PROJECT', # alias
               wflbucket_url='s3://WFLBUCKET/PIPELINE/VERSION',
```

Reply:
changed